packages/lodestar-validator/src/services/attestation.ts

```diff
@@ -138,39 +139,16 @@ export class AttestationService {
     this.logger.info(
       `Aggregating attestations for committee ${duty.committeeIndex} at slot ${duty.attestationSlot}`
     );
-    const wireAttestations = await this.provider.validator.getWireAttestations(
-      computeEpochAtSlot(this.config, duty.attestationSlot),
-      duty.committeeIndex
-    );
-    if(wireAttestations.length === 0) {
-      this.logger.warn(
-        `No attestations to aggregate for slot ${duty.attestationSlot} and committee ${duty.committeeIndex}`
-      );
-    }
-
-    const compatibleAttestations = wireAttestations.filter((wireAttestation) => {
-      return this.config.types.AttestationData.equals(wireAttestation.data, attestation.data)
-                    // prevent including aggregator attestation twice
-                    && ! this.config.types.Attestation.equals(attestation, wireAttestation);
-    });
-    compatibleAttestations.push(attestation);
-    const aggregatedAttestation: Attestation = {
-      signature: aggregateSignatures(compatibleAttestations.map((a) => a.signature.valueOf() as Uint8Array)),
-      data: attestation.data,
-      aggregationBits: compatibleAttestations.reduce((aggregatedBits, current) => {
-        for (let i = 0; i < aggregatedBits.length; i++) {
-          aggregatedBits[i] = aggregatedBits[i] || current.aggregationBits[i];
-        }
-        return aggregatedBits;
-      }, Array.from({length: attestation.aggregationBits.length}, () => false)),
-    };
+    const aggregatedAttestation = await this.provider.validator.produceAggregatedAttestation(attestation.data);
     await this.provider.validator.publishAggregatedAttestation(
-      aggregatedAttestation,
-      this.publicKey,
-      this.getSlotSignature(
-        duty.attestationSlot,
-        fork
-      )
+      {
+        aggregate: aggregatedAttestation,
+        selectionProof: this.getSlotSignature(
+          duty.attestationSlot,
+          fork
+        ),
+        aggregatorIndex: (await (this.provider.beacon.getValidator(this.publicKey))).index
```

Comment:
The index should likely be fetched and stored prior to this. It can be retrieved once, it will never change.

Reply:
I agree but I would wait with such optimization until we have proper ways to deal with validator status(we might not be able to fetch index on start if validator deposit isn't yet detected)

Reply:
maybe another stream api? A validator UI would need it to display correct status.

Reply:
Yes, but instead of introducing our own statuses and api, I would wait till we have some agreement on that